Exception thrown when native assembly is in assemblies component in extension manifest

## Description of bug
The `AssemblyInstaller` reads version information from provided assemblies to create a binding redirect for any strong-named assemblies.  The code that does this does not have error handling when there are errors reading that information, such as when the assembly file is a native assembly, rather than a .Net assembly.

## Steps to reproduce
1. Create an extension that packages a native assembly, e.g. [bad-assembly-image-demo.zip](https://github.com/dnnsoftware/Dnn.Platform/files/2903010/bad-assembly-image-demo.zip)
2. Install the extension via the Persona Bar, click through the wizard steps

## Current result
An error in the installation log
>Could not load file or assembly 'Microsoft.DiaSymReader.Native.amd64.dll' or one of its dependencies. The module was expected to contain an assembly manifest.

## Expected result
The assembly is installed without any error

## Screenshots
![error message screenshot](https://user-images.githubusercontent.com/59507/53374819-2344ed80-391f-11e9-92a7-ad9a505e3046.png)

## Error log
>17:01:44.382-06:00 [etg63][D:4][T:37][ERROR] DotNetNuke.Services.Installer.Log.Logger - Exception - Could not load file or assembly 'Microsoft.DiaSymReader.Native.amd64.dll' or one of its dependencies. The module was expected to contain an assembly manifest

## Affected version
<!-- Check all that apply and add more if necessary -->

* [x] 9.3.0
* [x] 9.2.2
* [x] 9.2.1
* [x] 9.2
* [x] 9.1.1
* [x] 9.1
* [x] 9.0

--- a/DNN Platform/Library/Services/Installer/Installers/AssemblyInstaller.cs
+++ b/DNN Platform/Library/Services/Installer/Installers/AssemblyInstaller.cs
@@ -23,11 +23,13 @@
 using System;
 using System.IO;
 using System.Reflection;
+using System.Security;
 using System.Text.RegularExpressions;
 using System.Xml;
 
 using DotNetNuke.Common;
 using DotNetNuke.Data;
+using DotNetNuke.Framework;
 
 #endregion
 
@@ -43,7 +45,9 @@ namespace DotNetNuke.Services.Installer.Installers
     {
         private static readonly Regex PublicKeyTokenRegex = new Regex(@"PublicKeyToken=(\w+)", RegexOptions.CultureInvariant | RegexOptions.Compiled);
 
-		#region "Protected Properties"
+        private static readonly string OldVersion = "0.0.0.0-" + new Version(short.MaxValue, short.MaxValue, short.MaxValue, short.MaxValue);
+
+        #region "Protected Properties"
 
         /// -----------------------------------------------------------------------------
         /// <summary>
@@ -221,29 +225,70 @@ namespace DotNetNuke.Services.Installer.Installers
             }
         }
 
-        /// <summary>If tyhe <paramref name="file"/> is a strong-named assembly, applies the XML merge.</summary>
+        /// <summary>If the <paramref name="file"/> is a strong-named assembly, applies the XML merge.</summary>
         /// <param name="file">The assembly file.</param>
         /// <param name="xmlMergeFile">The XML merge file name.</param>
-        /// <returns><c>true</c> if the XML Merge was applied successfully, <c>false</c> if the file was not a strong-named assembly.</returns>
+        /// <returns><c>true</c> if the XML Merge was applied successfully, <c>false</c> if the file was not a strong-named assembly or could not be read.</returns>
         private bool ApplyXmlMerge(InstallFile file, string xmlMergeFile)
         {
-            var assemblyName = AssemblyName.GetAssemblyName(Path.Combine(this.PhysicalBasePath, file.FullName));
-            if (!assemblyName.Flags.HasFlag(AssemblyNameFlags.PublicKey))
+            var assemblyName = ReadAssemblyName(Path.Combine(this.PhysicalBasePath, file.FullName));
+            var publicKeyToken = ReadPublicKey(assemblyName);
+            if (string.IsNullOrEmpty(publicKeyToken))
             {
                 return false;
             }
 
             var name = assemblyName.Name;
-            var publicKeyToken = PublicKeyTokenRegex.Match(assemblyName.FullName).Groups[1].Value;
-            var oldVersion = "0.0.0.0-" + new Version(short.MaxValue, short.MaxValue, short.MaxValue, short.MaxValue);
-            var newVersion = assemblyName.Version.ToString();
+            var assemblyVersion = assemblyName.Version;
+            var newVersion = assemblyVersion.ToString();
 
             var xmlMergePath = Path.Combine(Globals.InstallMapPath, "Config", xmlMergeFile);
-            var xmlMergeDoc = GetXmlMergeDoc(xmlMergePath, name, publicKeyToken, oldVersion, newVersion);
+            var xmlMergeDoc = GetXmlMergeDoc(xmlMergePath, name, publicKeyToken, OldVersion, newVersion);
             var xmlMerge = new XmlMerge(xmlMergeDoc, file.Version.ToString(), this.Package.Name);
             xmlMerge.UpdateConfigs();
             
             return true;
+        }
+
+        /// <summary>Reads the file's <see cref="AssemblyName"/>.</summary>
+        /// <param name="assemblyFile">The path for the assembly whose <see cref="AssemblyName"/> is to be returned.</param>
+        /// <returns>An <see cref="AssemblyName"/> or <c>null</c></returns>
+        private static AssemblyName ReadAssemblyName(string assemblyFile)
+        {
+            try
+            {
+                return AssemblyName.GetAssemblyName(assemblyFile);
+            }
+            catch (BadImageFormatException)
+            {
+                // assemblyFile is not a valid assembly.
+                return null;
+            }
+            catch (ArgumentException)
+            {
+                // assemblyFile is invalid, such as an assembly with an invalid culture.
+                return null;
+            }
+            catch (SecurityException)
+            {
+                // The caller does not have path discovery permission.
+                return null;
+            }
+            catch (FileLoadException)
+            {
+                // An assembly or module was loaded twice with two different sets of evidence.
+                return null;
+            }
+        }
+        
+        private static string ReadPublicKey(AssemblyName assemblyName)
+        {
+            if (assemblyName == null || !assemblyName.Flags.HasFlag(AssemblyNameFlags.PublicKey))
+            {
+                return null;
+            }
+
+            return PublicKeyTokenRegex.Match(assemblyName.FullName).Groups[1].Value;
         }
 
         /// <summary>Gets the XML merge document to create the binding redirect.</summary>

